semtypes/src/main/java/io/ballerina/semtype/UniformTypeOps.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.semtype;
+
+public interface UniformTypeOps {
+    SubtypeData union(SubtypeData t1, SubtypeData t2);
+    SubtypeData intersect(SubtypeData t1, SubtypeData t2);
+    SubtypeData diff(SubtypeData t1, SubtypeData t2);
+    SubtypeData complement(SubtypeData t);
+    boolean isEmpty(TypeCheckContext tc, SubtypeData t);
```

Comment:
I suspect it will work out better to have an interface with just the first 4 and then have another interface that adds isEmpty. Then you can have an abstract class that implements the first interface for Bdd.

Reply:
We already have a similar hierarchy here (just without interface segregation) https://github.com/ballerina-platform/ballerina-lang/blob/9cb56bbe2640a68c7a3124f8a53a04a04cfd6245/semtypes/src/main/java/io/ballerina/semtype/typeops/CommonOps.java#L23

Reply:
Oh please disregards the comment, I get what you meant.